doc/feature/spikeReport.md

```diff
@@ -0,0 +1,450 @@
+whose interface doesn't differ very much from a potential write interface in
+Spikes.
+
+### 3: Why the access time of the spikes container is amortized constant time instead of constant time?
+
+_Resolved: Yes_
+
+The rationale for amortized time is that it allows the implementation to use
+data structures different from a single flat memory buffer per container. With
+special data structures it's possible to implement zero-copy optimizations,
+smarter internal storage and even internal threading while still ensuring
+const-correctness. For example, imagine we have some client code that uses
+OpenMP to speed up the processing of a report:
+
+    #pragma omp parallel for shared(report)
+    for (int i = 0; i != slices; ++i)
+    {
+        auto slice = getTimeSlice(i);
+        const brion::Spikes& spikes = report.getSpikes(slice.first,
+                                                       slice.second);
+        ... // Do something
+    }
+
+With strict constant time, the previous code cannot return any reference to
+shared data from getSpikes because it may be relocated by subsequent calls from
+other threads, hence the only sane alternative is to return a copy of the data.
+
+By relaxing the time complexity requirements, it would be possible to use a
+lunchbox::lfVector for example.
+
+### 4: Can getSpikes return _const Spikes&_?
+
+_Resolved: Yes_
+
+No. That will imply that the container needs to be stored inside SpikeReport.
+As a side-effect this requires Spikes to be copyable and the copies need to be
+efficient, but thanks to issues #1 and #2 the implementation can provide
+cheap copies by sharing.
+
+### 4: Do you need a method to erase portions of the report?
+
+_Resolved: Partially_
+
+We don't have a use case for this yet, but it looks necessary for out-of-core
+processing of very large reports. The mail goal for this method in the original
+class was to clear the internal caching. Clearing the report can be very
+complex without requiring duplication of data. The compromise solution is to
+provide a method to clear the cache enterily. The implementation can easily
+keep Spike containers returned by getSpikes valid with smart pointers.
+
+### 6: For stream based reports should we block on getSpikes() or in the iterator ++ operator.
+
+_Resolved: Yes_
+
+It seems easier for the implementation to block on getSpikes than on
+the operator increment. Blocking in the container requires synchronization
+primitives to ensure const-correctness that wouldn't be otherwise needed.
+Locking at the iterator increment requires having a separate function to
+indicate the timeout, which is awkward, or not having a timeout at all for
+an equivalent to waitUntil and also add the current functions to know until
+which time is it possible to advance without blocking.
+
+### 7: Why getStartTime() retuns 0 for stream based reports, but it can be undefined for empty file reports?
+
+_Resolved: No_
+
+Seems reasonable, but it's open to discussion.
```

Comment:
In file based reports an empty report has UNDEFINED start and end times and that can't change over the course of time, so the client knows that there isn't anything to do with that. In stream reports we need to differentiate very clearly a finished stream from an ongoing one. If no spike data has been received and we make start time to return UNDEFINED that can be confused with the empty file report (unless we add isStream(), but you don't like that, do you?)